milestones/data.py

```diff
@@ -47,17 +47,132 @@ def _get_milestone_relationship_type(relationship):
         )
 
 
+def _activate_record(record):
+    """
+    Enables database records by setting the 'active' attribute to True
+    The queries in this module filter out inactive records as part of their criteria
+    This effectively allows us to soft-delete records so they are not lost forever
+    """
+    record.active = True
+    record.save()
+
+
 def _inactivate_record(record):
     """
     Disables database records by setting the 'active' attribute to False
     The queries in this module filter out inactive records as part of their criteria
     This effectively allows us to soft-delete records so they are not lost forever
     """
-
     record.active = False
     record.save()
 
 
+def _activate_milestone(milestone):
+    """
+    Activates an inactivated (soft-deleted) milestone as well as any inactive relationships
+    """
+    [_activate_course_milestone_relationship(record) for record
+     in internal.CourseMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_course_content_milestone_relationship(record) for record
+     in internal.CourseContentMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_user_milestone_relationship(record) for record
+     in internal.UserMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_record(record) for record
+     in internal.Milestone.objects.filter(id=milestone.id, active=False)]
```

Comment:
If it's all done in the space of a request and it's in-proc with edx-platform, it'll all be wrapped in an implicit transaction since we run transaction middleware. If we want to be explicit and mark certain methods in this way, we can, but the going assumption is 1 user request == 1 transaction.